File: primes2/getprime.c
#include "getprime.h"


int prime(int num) {
    if(num < 2)
        return 0;
    if(num == 2)
        return 1;
    if(num%2 == 0)
        return 0;

    for(int i = 3; i<(num/2); i += 2) {
        if(num%i == 0)
            return 0;
    }
    return 1;
}


int yieldprime(int target) {
    int count = 1;
    int primes = 2;
    while(1) {
        if(count == target)
            return primes;
        primes += 1;
        if(prime(primes))
            count++;
    }
}

File: python's-enhancement/primes3/getprime.py
#!/usr/bin/python3


import sys

try:
    import _getprime
except ImportError:
    print('Need build extension with python3 setup.py build_ext --inplace')
    sys.exit(1)

from time import time


def main():
    start = time()

    try:
        target = int(sys.argv[1])
        if target <= 0:
            raise ValueError
    except(IndexError, ValueError):
        print('Needed a positive integer as command line argument')
        return

    prime = _getprime.yieldprime(target)

    print('Taked %.12f seconds to fetch' %(time() - start))
    print(prime)


if __name__=='__main__':
    main()
